lib/internal/locks.js

```diff
@@ -0,0 +1,295 @@
+   * @param {string} name - The name of the lock resource
+   * @param {object} [options] - Lock options (optional)
+   * @param {string} [options.mode='exclusive'] - Lock mode: 'exclusive' or 'shared'
+   * @param {boolean} [options.ifAvailable=false] - Only grant if immediately available
+   * @param {boolean} [options.steal=false] - Steal existing locks with same name
+   * @param {AbortSignal} [options.signal] - Signal to abort pending lock request
+   * @param {Function} callback - Function called when lock is granted
+   * @returns {Promise} Promise that resolves when the lock is released
+   * @throws {TypeError} When name is not a string or callback is not a function
+   * @throws {DOMException} When validation fails or operation is not supported
+   */
+  // https://w3c.github.io/web-locks/#api-lock-manager-request
+  async request(name, options, callback) {
+    if (callback === undefined) {
+      callback = options;
+      options = undefined;
+    }
+
+    name = converters.DOMString(name);
+    validateFunction(callback, 'callback');
+
+    if (options === undefined || typeof options === 'function') {
+      options = kEmptyObject;
+    }
+
+    // Convert LockOptions dictionary
+    options = convertLockOptions(options);
+
+    const { mode, ifAvailable, steal, signal } = options;
+
+    validateAbortSignal(signal, 'options.signal');
+
+    if (name.startsWith('-')) {
+      // If name starts with U+002D HYPHEN-MINUS (-), then reject promise with a
+      // "NotSupportedError" DOMException.
+      throw lazyDOMException('Lock name may not start with hyphen',
+                             'NotSupportedError');
+    }
+
+    if (ifAvailable === true && steal === true) {
+      // If both options' steal dictionary member and option's
+      // ifAvailable dictionary member are true, then reject promise with a
+      // "NotSupportedError" DOMException.
+      throw lazyDOMException('ifAvailable and steal are mutually exclusive',
+                             'NotSupportedError');
+    }
+
+    if (mode !== locks.LOCK_MODE_EXCLUSIVE && steal === true) {
+      // If options' steal dictionary member is true and options' mode
+      // dictionary member is not "exclusive", then return a promise rejected
+      // with a "NotSupportedError" DOMException.
+      throw lazyDOMException(`mode: "${locks.LOCK_MODE_SHARED}" and steal are mutually exclusive`,
+                             'NotSupportedError');
+    }
+
+    if (signal && (steal === true || ifAvailable === true)) {
+      // If options' signal dictionary member is present, and either of
+      // options' steal dictionary member or options' ifAvailable dictionary
+      // member is true, then return a promise rejected with a
+      // "NotSupportedError" DOMException.
+      throw lazyDOMException('signal cannot be used with steal or ifAvailable',
+                             'NotSupportedError');
+    }
+
+    if (signal) {
+      signal.throwIfAborted();
+    }
```

Comment:
Nit: I'd suggest moving this up immediately after the `validateAbortSignal(...)` call ... or, move it into the `if (signal) {` block below.